sklearn/ensemble/gradient_boosting.py

```diff
@@ -1367,9 +1381,12 @@ def staged_predict_proba(self, X):
         try:
             for score in self.staged_decision_function(X):
                 yield self.loss_._score_to_proba(score)
-        except AttributeError:
-            raise AttributeError('loss=%r does not support predict_proba' %
-                                 self.loss)
+        except AttributeError as e:
+            if isinstance(e, NotFittedError):
+                raise e
+            else:
+                raise AttributeError('loss=%r does not support predict_proba' %
+                                     self.loss)
```

Comment:
Line 1347 / 1358 doe not mention ``e``. I don't understand why ``e`` is defined in line 1359.